src/arviz_plots/plots/utils.py

```diff
@@ -295,3 +295,113 @@ def add_reference_lines(
             **ref_kwargs,
         )
     return plot_collection
+
+
+def add_reference_bands(
+    plot_collection,
+    references,
+    orientation="vertical",
+    aes_map=None,
+    plot_kwargs=None,
+    sample_dims=None,
+    ref_dim=None,
+    **kwargs,
+):
+    """Add reference bands.
+
+    This function adds lines to a plot collection based on the provided
+    references. It supports both vertical and horizontal lines, depending on the
+    specified orientation.
+
+    Parameters
+    ----------
+    plot_collection : PlotCollection
+        Plot collection to which the reference lines will be added.
+    references : tuple, list or dict
+        Reference values to be plotted as bands/shaded regions.
+    orientation : str, default "vertical"
+        The orientation of the reference lines, either "vertical" or "horizontal".
+    aes_map : mapping of {str : sequence of str}, optional
+        Mapping of artists to aesthetics that should use their mapping in `plot_collection`
+        when plotted. Valid keys are the same as for `plot_kwargs`.
+
+        The default is to use an "overlay_band" aesthetic for all elements.
+
+        It is possible to request aesthetics without mappings defined in the
+        provided `plot_collection`. In those cases, a mapping of "ref_dim" to the requested
+        aesthetic will be automatically added.
+    plot_kwargs : mapping of {str : mapping or False}, optional
+        Valid keys are:
+
+        * "ref_band" -> passed to :func:`~arviz_plots.visuals.vspan` for vertical `orientation`
+          and to :func:`~arviz_plots.visuals.hspan` for horizontal `orientation`
+
+    sample_dims : list, optional
+        Dimensions to reduce unless mapped to an aesthetic.
+        Defaults to ``rcParams["data.sample_dims"]``
+    ref_dim : list, optional
+        Specifies the name of the reference dimension for reference values.
+    **kwargs : mapping of {str : sequence}, optional
```

Comment:
```suggestion
    **kwargs : sequence, optional
```

This one I recently learned I was doing wrong. The fact it is `**kwargs` and not `kwargs` means it must be a mapping with strings as keys, so in such cases only the type of the values should be documented.